src/pudl/transform/ferc1.py

```diff
@@ -1605,6 +1861,84 @@ def _clean_cols(df, table_name):
     return df
 
 
+def add_report_year_column(df):
+    """Add a report date column."""
+    df = df.assign(report_year=lambda x: pd.to_datetime(x.start_date).dt.year)
+    return df
+
+
+def assign_record_id(
+    df: pd.DataFrame, table_name: TABLES_LITERAL, source_ferc1: Literal["dbf", "xbrl"]
+) -> pd.DataFrame:
+    """Assign a record ID column from either XBRL or DBF ferc1 source table.
+
+    It is often useful to be able to tell exactly which record in the FERC Form 1
+    database a given record within the PUDL database came from.
+
+    Within each FERC Form 1 DBF table, each record is supposed to be uniquely
+    identified by the combination of:
+    report_year, report_prd, respondent_id, spplmnt_num, row_number.
+
+    The FERC Form 1 XBRL tables do not have these supplement and row number
+    columns, so we construct an id based on:
+    report_year, entity_id, and the primary key columns of the XBRL table
+
+    Args:
+        df: table to assign `record_id` to
+        table_name: name of table
+        source_ferc1: data source of raw ferc1 database.
+
+    Raises:
+        AssertionError: If the resulting `record_id` column is non-unique.
+    """
+    if source_ferc1 == "xbrl":
+        pk_cols = TABLE_PKS_XBRL[table_name]
+    elif source_ferc1 == "dbf":
+        pk_cols = [
+            "report_year",
+            "report_prd",
+            "respondent_id",
+            "spplmnt_num",
+            "row_number",
+        ]
+        # for the dbf years, we used the original "f1_" table names for the id
+        # table_name = pudl_tables_to_dbf_tables.get(table_name)
+    else:
+        raise ValueError(
+            f"source_ferc1 must be either `xbrl` or `dbf`. Got {source_ferc1}"
+        )
+    if df[pk_cols].isnull().any().any():
+        raise AssertionError(
+            f"Null field found in ferc1 table {table_name}. \n{df[pk_cols].isnull().any()}"
+        )
+
+    df = df.assign(
+        temp=table_name,
+        record_id=lambda x: x.temp.str.cat(x[pk_cols].astype(str), sep="_"),
+    ).drop(columns=["temp"])
+
+    df = pudl.helpers.simplify_strings(df, ["record_id"])
+    if df.record_id.duplicated().any():
+        raise AssertionError(
+            "Record id column cannot result in duplicates. Columns are not "
+            f"unique primary keys: {pk_cols}. \nNon-unqiue records:\n"
+            f"{df[df.record_id.duplicated()][['record_id'] + pk_cols]}"
+        )
+    return df
+
+
+def assign_utility_id_ferc1_xbrl(df: pd.DataFrame) -> pd.DataFrame:
+    """Assign utility_id_ferc1.
+
+    This is a shell rn. Issue #1705
+    """
+    return df.assign(
+        utility_id_ferc1=lambda x: x.entity_id.str.replace("C", "")
+        .str.lstrip("0")
+        .astype("Int64")
```

Comment:
i needed to have a unique ID for the plant-id-er so i did this. this is obviously a hack and should be supplanted w/ a true solution from #1705